test/Microsoft.AspNet.Mvc.FunctionalTests/LinkGenerationTests.cs

```diff
@@ -45,8 +46,10 @@ public class LinkGenerationTests
 
             // Assert
             Assert.Equal(HttpStatusCode.Redirect, response.StatusCode);
-
-            Assert.Equal(expected, response.Headers.Location.ToString());
+            // Location.ToString() in mono returns file://url. (https://github.com/aspnet/External/issues/21)
+            Assert.Equal(
+                TestPlatformHelper.IsMono ? new Uri(expected) : new Uri(expected, UriKind.Relative),
```

Comment:
Mono seems to add  `file://` prefix to all URIs except the ones created with the above overload. So not changing it.